akka-stream/src/main/scala/DocumentStage.scala

```diff
@@ -21,7 +21,7 @@ private[akkastream] class DocumentStage[T](
     maxDocs: Int,
     err: ErrorHandler[Option[T]]
 )(implicit ec: ExecutionContext)
-  extends GraphStage[SourceShape[T]] {
+  extends GraphStageWithMaterializedValue[SourceShape[T], Future[State]] {
```

Comment:
To keep compatibility would require to implement also `GraphStage` with `createLogic` as `@deprecated` using `createLogicAndMaterializedValue` (ignoring the mat val)

Reply:
Do we really want to keep compatibility for package private api? Can we just filter out those classes in mima settings?

Reply:
If the deprecation option would have a to high cost, it could be filtrer, but in such case that's quite simple

Reply:
Looks like MiMa filtering will be required for this one: `GraphStage` already extends `GraphStageWithMaterializedValue` with a `final` override of `createLogicAndMaterializedValue`.